### ISSUE TITLE
Unable to download chat files on Mac with CLI

### ISSUE THREAD

**Issue by YousufSSyed** [lay295/TwitchDownloader#207]
Here's a past livestream:
https://www.twitch.tv/videos/1107963619
and my command:
`./TwitchDownloaderCLI -m ChatDownload --timestamp --embed-emotes -u 1107963619 -o "Vesteria Dev Mushtown Improvements.json"`
and my output:
```
Failure processing application bundle.
Failed to determine location for extracting embedded files
DOTNET_BUNDLE_EXTRACT_BASE_DIR is not set, and a read-write cache directory couldn't be created.
A fatal error was encountered. Could not extract contents of the bundle
```
(if the livestream gets deleted, you can reproduce this with another past livestream)
I'm on Big Sur 11.5.1. The command (without `./`) works on my Windows PC.

**Comment by GTO683** [comment]
> Here's a past livestream:
> 
> https://www.twitch.tv/videos/1107963619
> 
> and my command:
> 
> `./TwitchDownloaderCLI -m ChatDownload --timestamp --embed-emotes -u 1107963619 -o "Vesteria Dev Mushtown Improvements.json"`
> 
> and my output:
> 
> ```
> 
> Failure processing application bundle.
> 
> Failed to determine location for extracting embedded files
> 
> DOTNET_BUNDLE_EXTRACT_BASE_DIR is not set, and a read-write cache directory couldn't be created.
> 
> A fatal error was encountered. Could not extract contents of the bundle
> 
> ```
> 
> (if the livestream gets deleted, you can reproduce this with another past livestream)
> 
> I'm on Big Sur 11.5.1. The command (without `./`) works on my Windows PC.

Please teach me how to set up and use the cli with terminal on mac no clear instructions anywhere.

**Comment by lay295** [comment]
@YousufSSyed are you perhaps on an M1 Mac? A few months ago I tried on a regular Intel Mac and it worked fine. I don't have a Mac currently to debug or test



> Please teach me how to set up and use the cli with terminal on mac no clear instructions anywhere.

Have you looked at [this](https://github.com/lay295/TwitchDownloader/tree/master/TwitchDownloaderCLI)?

**Comment by GTO683** [comment]
> @YousufSSyed are you perhaps on an M1 Mac? A few months ago I tried on a regular Intel Mac and it worked fine. I don't have a Mac currently to debug or test
> 
> 
> 
> 
> 
> 
> 
> > Please teach me how to set up and use the cli with terminal on mac no clear instructions anywhere.
> 
> 
> 
> Have you looked at [this](https://github.com/lay295/TwitchDownloader/tree/master/TwitchDownloaderCLI)?

I noticed the arguments glossary but how do I set it up first? Im not too experienced with code. I unzipped the CLI and copy and pasted one of your examples (and replaced Id number of course) in terminal and it said it didn't recognize "TwitchDownloaderCLI"

**Comment by YousufSSyed** [comment]
I'm using a 2015 MacBook Pro (not M1). 

> I noticed the arguments glossary but how do I set it up first? Im not too experienced with code. I unzipped the CLI and copy and pasted one of your examples in that section (and replaced Id number of course) in terminal and it said it didn't recognize "TwitchDownloaderCLI"

I'm working a written tutorial for archiving Twitch streams, including using the TwitchDownloaderCLI on mac. Refer to this: https://minershaven.fandom.com/wiki/User:TheWikiaEditMachine/Twitch_Archiving/Creating_The_Videos#Mac_CLI_Setup

**Comment by GTO683** [comment]
> I'm using a 2015 MacBook Pro (not M1). 
> 
> 
> 
> > I noticed the arguments glossary but how do I set it up first? Im not too experienced with code. I unzipped the CLI and copy and pasted one of your examples in that section (and replaced Id number of course) in terminal and it said it didn't recognize "TwitchDownloaderCLI"
> 
> 
> 
> I'm working a written tutorial for archiving Twitch streams, including using the TwitchDownloaderCLI on mac. Refer to this: https://minershaven.fandom.com/wiki/User:TheWikiaEditMachine/Twitch_Archiving/Creating_The_Videos#Mac_CLI_Setup
> 
> 
> 
> 

Im sorry, what is the "work directory" and where is it located? Noob here. Its mentioned in 2 steps and I'm not sure what it means. "Download TwitchDownloaderCLI here (TwitchDownloaderCLI-MacOS-x64.zip) **_and place it in your directory_**. And, "**_With Terminal in your work directory_**, you must first give the terminal permission to run it." Plz explain.

**Comment by lay295** [comment]
I installed a MacOS BigSur VM on one of my machines, and I cannot reproduce the issue. When I had physical access to a real Mac and tried the program it worked as well.

https://www.youtube.com/watch?v=pqWaYH9XdVQ

This seems to possibly be related to [this issue](https://github.com/dotnet/runtime/issues/3846) maybe? I don't know why .NET Core is failing to extract the bundle, but looking around it seems the only way to fix it is to set the DOTNET_BUNDLE_EXTRACT_BASE_DIR environment variable. Perhaps it might fix itself when I eventually target a newer .NET framework, but I don't think this is something I can fix myself and doesn't seem to be a very common issue

@GTO683 Watch the video I posted for an example of how to run it.

Basically.
Download TwitchDownloaderCLI-MacOS-x64.zip to your Downloads folder
Extract TwitchDownloaderCLI-MacOS-x64.zip to your Downloads folder
Open Terminal
Type "cd Downloads" and hit enter.
Type "chmod +x TwitchDownloaderCLI" and hit enter.

Then you can run commands with ./TwitchDownloaderCLI, if it doesn't let you run it you have to allow it in the settings like in the video.

**Comment by GTO683** [comment]
> I installed a MacOS BigSur VM on one of my machines, and I cannot reproduce the issue. When I had physical access to a real Mac a few months ago and tried the program it worked as well.
> 
> 
> 
> https://www.youtube.com/watch?v=pqWaYH9XdVQ
> 
> 
> 
> This seems to possibly be related to [this issue](https://github.com/dotnet/runtime/issues/3846) maybe? I don't know why .NET Core is failing to extract the bundle, but looking around it seems the only way to fix it is to set the DOTNET_BUNDLE_EXTRACT_BASE_DIR environment variable. Perhaps it might fix itself when I eventually target a newer .NET framework, but I don't think this is something I can fix myself and doesn't seem to be a very common issue
> 
> 
> 
> @GTO683 Watch the video I posted for an example of how to run it.
> 
> 
> 
> Basically.
> 
> Download TwitchDownloaderCLI-MacOS-x64.zip to your Downloads folder
> 
> Extract TwitchDownloaderCLI-MacOS-x64.zip to your Downloads folder
> 
> Open Terminal
> 
> Type "cd Downloads" and hit enter.
> 
> Type "chmod +x TwitchDownloaderCLI" and hit enter.
> 
> 
> 
> Then you can run commands with ./TwitchDownloaderCLI, if it doesn't let you run it you have to allow it in the settings like in the video.

So I followed exactly what you did in the video however, I dont have a prompt in "Security & Privacy" settings to allow it anyway. I just get `zsh: permission denied: TwitchDownloderCLI` when typing ./TwitchDownloderCLI in terminal with no option in settings. And its weird because I usually get that prompt when downloading apps online but not when I typed the terminal command. What do you recommend? Should I follow the steps in the link @YousufSSyed posted? Would that make a difference? Is this a dangerous way? [Here they are](https://user-images.githubusercontent.com/79897652/129447161-cd4f3b48-49d1-45ca-8603-4c7b4fedddc6.jpeg). Bare with me I'm a noob. Just trying to be as specific as possible.

**Comment by GTO683** [comment]
Wait! I attempted it again, and it worked! The prompt showed and I allowed it! Thanks @lay295

**Comment by GTO683** [comment]
Now when I attempt to download a vod I get zsh: command not found: TwitchDownloaderCLI

**Comment by GTO683** [comment]
Question: do arguments have to be listed in a specific order?

**Comment by YousufSSyed** [comment]
Sorry I've been busy all day but you just needed to ignore the phrase work directory and just put the files wherever you wanted. But I'm glad you got it working! Also, I think args don't need to be in order, I haven't had issues with it not being in any order.

**Comment by GTO683** [comment]
Yea no worries man thx for helping. I tried to download a vod and it worked but when I tried to download a clip later it said twitchdownloadercli no such file or directory. I think CLIs are just too complicated for me lol. > Sorry I've been busy all day but you just needed to ignore the phrase work directory and just put the files wherever you wanted. But I'm glad you got it working! Also, I think args don't need to be in order, I haven't had issues with it not being in any order.

**Comment by YousufSSyed** [comment]
@lay295 I found that rendering also doesn't work and produces the exact same output. Here's the command I tried:
`TwitchDownloaderCLI -m ChatRender --sub-messages --bttv true --ffz true --timestamp --framerate 60 --chat-height 1080 --chat-width 426 -i "Vesteria Dev Mushtown Improvements.json" -o "Vesteria Dev Mushtown Improvements Chat.mp4"`
Should this be a separate issue?

**Comment by ScrubN** [comment]
> I'm working a written tutorial for archiving Twitch streams, including using the TwitchDownloaderCLI on mac. Refer to this: https://minershaven.fandom.com/wiki/User:TheWikiaEditMachine/Twitch_Archiving/Creating_The_Videos#Mac_CLI_Setup

@YousufSSyed Your guide is cool but sadly now also out of date. For example, `--timestamp` is no longer a valid arg, the Inter font is included, and the look of the GUI has changed significantly.

> I can confirm, I am running on macOS M1 (Big Sur) and I am currently unable to render a downloaded chat, following this command for instance:

@tikodeur there was an issue regarding MacOS library bundling which has since been fixed. If further issues continue please open a new issue
